docs/ext/commands/additional_info.rst

```diff
@@ -32,3 +32,28 @@ If your bot requires shard distribution across several machines, you should set
 This will prevent conflicts and race conditions. Discord API doesn't provide users with events related to application command updates,
 so it's impossible to keep the cache of multiple machines synced. Having only 1 machine with ``sync_commands`` set to ``True`` is enough
 because global registration of application commands doesn't depend on sharding.
+
+.. _why_params_and_injections_return_any:
+
+Why ``Param`` and ``Injection``-related functions return ``Any``?
+-----------------------------------------------------------------
+
+If your editor of choice supports type-checking, you may have noticed that :func:`~ext.commands.Param`, :func:`~ext.commands.inject`,
+:func:`~ext.commands.injection` and :func:`~ext.commands.register_injection` do not have a specific return type, but at runtime these return :class:`~ext.commands.Injection`.
+
+A typical example of slash command might look like this: ::
+
+    @bot.slash_command(description="Replies with the given text!")
+    async def echo(
+        inter: disnake.ApplicationCommandInteraction,
+        text: str = commands.Param(description="Echo~")
+    ) -> None:
+        await inter.response.send_message(text)
+
+Here, you have two parameters in your command's function: ``inter``, an instance of :class:`disnake.ApplicationCommandInteraction`, and
+``text``, which is.. Noticed? You annotate ``text`` as ``str``, but at the same time, assign :class:`~ext.commands.ParamInfo` instance to it.
+That's the thing. When your editor type-checks your (any library's) code, it would normally complain if you tried to do the above, because
+you're trying to "convert" ``ParamInfo`` into ``str``, but since library have set ``Param``'s return type as ``Any``, type-checker "accepts"
+your code, because ``str`` (and any type) is the subset of ``Any``. Win-win!
+
+Same thing applies to ``Injection``-related functions.
```

Comment:
This line is redundant due to lines 41 and 42 listing all of the affected functions.

Reply:
I dont think so, lines you've mentioned just list all things that return Any, then the explanation goes, and then line 59 states that explanation above applies to these functions as well, that answer to "Why?" is the same

Reply:
Changed it in https://github.com/DisnakeDev/disnake/pull/675/commits/c2d003fa7921e345c28fa3acadbe4d3305fd907f